transaction/transaction.go

```diff
@@ -166,6 +166,13 @@ func (tx *Transaction) GetProgramHashes() ([]common.Uint160, error) {
 			return nil, err
 		}
 		hashes = append(hashes, programhash)
+	case pb.PayloadType_GENERATE_ID_2_TYPE:
+		sender := payload.(*pb.GenerateID2).Sender
+		programhash, err := program.CreateProgramHash(sender)
```

Comment:
This is wrong, you should use `common.BytesToUint160(sender)`. See other tx type with sender for example